### PR TITLE
New optional user info in Order payload

### PR DESCRIPTION
**Pull request by arkanoider**
@grunch @Catrya @AndreaDiazCorreia ,

this is a proposal for the upgrade in the message that maker receives after taker did his part, it's a bit different from what @grunch proposed, because it's just easier to add a UserInfo optional field in `Payload` enum like this:

```Rust
pub enum Payload {
    /// Order
   Order(SmallOrder, Option<UserInfo>),
    /// Payment request
    PaymentRequest(Option<SmallOrder>, String, Option<Amount>),
    /// Use to send a message to another user
    TextMessage(String),
    /// Peer information
    Peer(Peer),
    /// Used to rate a user
    RatingUser(u8),
    /// In some cases we need to send an amount
    Amount(Amount),
    /// Dispute
    Dispute(Uuid, Option<SolverDisputeInfo>),
    /// Here the reason why we can't do the action
    CantDo(Option<CantDoReason>),
    /// This is used by the maker of a range order only on
    /// messages with action release and fiat-sent
    /// to inform the next trade pubkey and trade index
    NextTrade(String, u32),
    /// Payment failure retry configuration information
    PaymentFailed(PaymentFailedInfo),
    /// Restore session data with orders and disputes
    RestoreData(RestoreSessionInfo),
    /// IDs array
    Ids(Vec<Uuid>),
    /// Orders array
    Orders(Vec<SmallOrder>),
}
```
This results in a Json struct like this:

```Json
{
  "order": {
    "version": 1,
    "request_id": 1,
    "trade_index": 2,
    "id": "308e1272-d5f4-47e6-bd97-3504baea9c23",
    "action": "new-order",
    "payload": {
      "order": [
        {
          "id": "308e1272-d5f4-47e6-bd97-3504baea9c23",
          "kind": "sell",
          "status": "pending",
          "amount": 100,
          "fiat_code": "eur",
          "min_amount": null,
          "max_amount": null,
          "fiat_amount": 100,
          "payment_method": "SEPA,Bank transfer",
          "premium": 1,
          "created_at": 1627371434,
          "expires_at": null
        },
        {
          "rating": 4.5,
          "reviews": 10,
          "operating_days": 30
        }
      ]
    }
  }
}
```
Is an easy task for mobile to decode this?

Second question, just asking for my understanding, @catrya scenario pasted on telegram seems related to `buy` order, in that case if:

- A (buyer) creates a buy order
- B (seller) takes it
- B (seller) Pays the invoice

At this point buyer has received the two messages of confirmation:
- one with the order
- one with user reputation

and in this case we want to make a single message of these two.

If we are in a `sell` order case and:

- A (seller) creates a sell order
- B (buyer) takes it
- B (buyer) Add the requested invoice

The seller now will find these two messages ( copying from mostro-cli):

```Bash

📄 Message 3:
─────────────────────────────────────
⏰ Time: 2025-10-26 17:10:41
📨 From: 🧌 acf3c926f37102b03a5ff8a83fa4480af59452e47d84e15513e5adfa6c2aac83
🎯 Action: ⚡ PayInvoice
📝 Details:
   ⚡ Lightning Invoice:
   lnbcrt31660n1p50u4s3pp5n7t3gpafq5yrjxs8apd8qzjex6avjka9cvqwm5dw78j8psqxnfeqdf5g4ekxun0wusxzmt0w4h8ggz0wfjx2u3qyv6kzwrxv9jnzwfdv5crvdedxs6n2epd8pjxyd3dvgek2wpjxdjnzwrxxq6r5gzng4xycgzz23pjqen0wgsy242jyqejqtfqf96zq46ff3xzq3jjg4z453fqf98zq46pf3xy24pwypyhggrhd9kxcgrjv4kx2ctnv5sx7mnrv5s8jmm4ypex2mr9v9ek2t3qf96zqamfd3kzqun9w36hym3qd9nzqcn409jhygrydajhxgrwda6zqcm0denxjundyp6xsefqwpshjmt9de6qcqzysxqyz5vqsp5mcnz2k9a58hr7s5kc8lra5uprwu0mp86w7uze024thhz4qnychyq9qxpqysgqnxlakfmvv0awz7ahp5ngnvauqd6nwewcxhjq95kpkpncmnqhs3c472zk9a63p45wmneug9c3tyz9wjeysxvfkf4gs0emza6fedrqgpqpuwdk26

📄 Message 4:
─────────────────────────────────────
⏰ Time: 2025-10-26 17:10:41
📨 From: 🧌 acf3c926f37102b03a5ff8a83fa4480af59452e47d84e15513e5adfa6c2aac83
🎯 Action: ⚡ PayInvoice
📝 Details:
   👤 Peer: Anonymous
   🔻 Rating: 0.0/5.0
   📊 Reviews: 0
   📅 Operating Days: 0
```
So also in this case we want to create a single message merging the invoice with user reputation? When seller pays invoice then he receives:

```Bash
📨 Direct Messages
═══════════════════════════════════════
📄 Message 1:
─────────────────────────────────────
⏰ Time: 2025-10-26 17:17:39
📨 From: 🧌 acf3c926f37102b03a5ff8a83fa4480af59452e47d84e15513e5adfa6c2aac83
🎯 Action: 🎯 BuyerTookOrder
📝 Details:
   📋 Order: 5a8fae19-e067-455d-8db6-b3e823e18f04 3074 sats (EUR)
   ✅ Status: Active
   📉 Kind: Sell
```

So here we again a payload with the order inside, but I think that reputation makes more sense in the previous step, not here.

Let me know your opinion before proceeding so we are aligned!

Ah last thing...are these the cases you have in mind @Catrya ?? Am I missing other cases?

## Summary by CodeRabbit

* **Enhancements**
  * Order payload now carries optional user information alongside order data.
  * Improved encryption key handling for safer cryptographic operations.

* **Tests**
  * Added comprehensive tests for Order messages with/without optional user info, including edge cases and round-trip validation.

* **Chores**
  * Updated Rust toolchain version.

### CODE REVIEW

File: src/message.rs [hunk 1]
```diff
@@ -320,7 +320,7 @@ pub struct RestoreSessionInfo {
 #[serde(rename_all = "snake_case")]
 pub enum Payload {
     /// Order
-    Order(SmallOrder),
+    Order(SmallOrder, Option<UserInfo>),
```
[the remark this answers is not in the document]
Comment (arkanoider) [reply src/message.rs]: Show me the differences with example json structs @coderabbitai
